lib/ansible/modules/cloud/docker/docker_config.py

```diff
@@ -0,0 +1,285 @@
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: docker_config
+
+short_description: Manage docker configs.
+
+version_added: "2.6"
+
+description:
+     - Create and remove Docker configs in a Swarm environment. Similar to `docker config create` and `docker config rm`.
+     - Adds to the metadata of new configs 'ansible_key', an encrypted hash representation of the data, which is then used
+     - in future runs to test if a config has changed.
+     - If 'ansible_key is not present, then a config will not be updated unless the C(force) option is set.
+     - Updates to configs are performed by removing the config and creating it again.
+options:
+  data:
+    description:
+      - String. The value of the config. Required when state is C(present).
+    required: false
+  labels:
+    description:
+      - "A map of key:value meta data, where both the I(key) and I(value) are expected to be a string."
+      - If new meta data is provided, or existing meta data is modified, the config will be updated by removing it and creating it again.
+    required: false
+  force:
+    description:
+      - Use with state C(present) to always remove and recreate an existing config.
+      - If I(true), an existing config will be replaced, even if it has not changed.
+    default: false
+    type: bool
+  name:
+    description:
+      - The name of the config.
+    required: true
+  state:
+    description:
+      - Set to C(present), if the config should exist, and C(absent), if it should not.
+    required: false
+    default: present
+    choices:
+      - absent
+      - present
+
+extends_documentation_fragment:
+    - docker
+
+requirements:
+  - "docker-py >= 2.6.0"
+  - "Docker API >= 1.30"
+
+author:
+  - Chris Houseknecht (@chouseknecht)
+  - John Hu (@ushuz)
+'''
+
+EXAMPLES = '''
+
+- name: Create config foo
+  docker_config:
+    name: foo
+    data: Hello World!
+    state: present
+
+- name: Change the config data
+  docker_config:
+    name: foo
+    data: Goodnight everyone!
+    labels:
+      bar: baz
+      one: '1'
+    state: present
+
+- name: Add a new label
+  docker_config:
+    name: foo
+    data: Goodnight everyone!
+    labels:
+      bar: baz
+      one: '1'
+      # Adding a new label will cause a remove/create of the config
+      two: '2'
+    state: present
+
+- name: No change
+  docker_config:
+    name: foo
+    data: Goodnight everyone!
+    labels:
+      bar: baz
+      one: '1'
+      # Even though 'two' is missing, there is no change to the existing config
+    state: present
+
+- name: Update an existing label
+  docker_config:
+    name: foo
+    data: Goodnight everyone!
+    labels:
+      bar: monkey   # Changing a label will cause a remove/create of the config
+      one: '1'
+    state: present
+
+- name: Force the removal/creation of the config
```

Comment:
`force: yes` does not help for removal, and besides the example says `state: present`, so please change this to `Force the (re-)creation of the config` (or similar).